### ISSUE TITLE
4.2: astropy.wcs.wcs.FITSFixedWarning: 'datfix' made the change 'Success' during tests

### ISSUE THREAD

**Issue by olebole** [astropy/astropy#11322]
### Description

I just tried to rebuild Astropy 4.2 for Debian, and stumbled in a number of errors and failures. Aside from the problems already reported, they are all connected to an `astropy.wcs.wcs.FITSFixedWarning: 'datfix' made the change 'Success'`.

This *may* be connected to an update of wcslib from 7.3.1 to 7.4 (maybe, in combination with the update of erfa from 1.7.1 to 1.7.2). Interestingly, the problem [didn't appear with just testing the binary Astropy package that was built with wcslib 7.3.1](https://ci.debian.net/data/autopkgtest/testing/amd64/a/astropy/10344902/log.gz), and the wcslib updated afterwards to 7.4 (should be possible, since they are claimed ABI compatibe), but it happens only when using the development package (header files?) of wcslib-7.4. That made the problem slip into Debian testing (otherwise, wcslib would have been blocked from the migration).

### Expected behavior

Tests should pass

### Actual behavior

One of the stacktraces:
```
_________________________________ test_hst_wcs _________________________________

    def test_hst_wcs():
        path = get_pkg_data_filename("data/dist_lookup.fits.gz")
    
        with fits.open(path) as hdulist:
            # wcslib will complain about the distortion parameters if they
            # weren't correctly deleted from the header
>           w = wcs.WCS(hdulist[1].header, hdulist)

astropy/wcs/tests/test_wcs.py:916: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/wcs/wcs.py:527: in __init__
    self.fix(translate_units=translate_units)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = WCS Keywords

Number of WCS axes: 2
CTYPE : 'RA---TAN-SIP'  'DEC--TAN-SIP'  
CRVAL : 5.63056810618  -72.05457184278998....29055156973602e-05  5.9525007864732e-06  
CD2_1 CD2_2  : 5.02263821027651e-06  -1.2644844123757e-05  
NAXIS : 100  100
translate_units = '', naxis = None

    def fix(self, translate_units='', naxis=None):
        """… """
        if self.wcs is not None:
            self._fix_scamp()
            fixes = self.wcs.fix(translate_units, naxis)
            for key, val in fixes.items():
                if val != "No change":
                    if (key == 'datfix' and '1858-11-17' in val and
                            not np.count_nonzero(self.wcs.mjdref)):
                        continue
>                   warnings.warn(
                        ("'{0}' made the change '{1}'.").
                        format(key, val),
                        FITSFixedWarning)
E                   astropy.wcs.wcs.FITSFixedWarning: 'datfix' made the change 'Success'.

astropy/wcs/wcs.py:706: FITSFixedWarning
```

### Steps to Reproduce

Install the packages from current Debian Testing, and try to build Astropy 4.2.

### System Details

* Astropy 4.2
* wcslib 7.4
* erfa 1.7.1

**Comment by olebole** [comment]
When I pragmatically change https://github.com/astropy/astropy/blob/a6347384ffeea47d062bee1f3ea7f60af793d95b/astropy/wcs/wcs.py#L703 to `if val not in ("No change", "Success"):`, I still get three WCS related failures:

```
___________________________ TestSpectra.test_spectra ___________________________

self = <astropy.wcs.tests.test_wcs.TestSpectra object at 0x7fe78b251490>

    def test_spectra(self):
        for filename in self._file_list:
            # use the base name of the file, so we get more useful messages
            # for failing tests.
            filename = os.path.basename(filename)
            # Now find the associated file in the installed wcs test directory.
            header = get_pkg_data_contents(
                os.path.join("data", "spectra", filename), encoding='binary')
            # finally run the test.
>           all_wcs = wcs.find_all_wcs(header)

astropy/wcs/tests/test_wcs.py:107: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/wcs/wcs.py:3348: in find_all_wcs
    subresult.fix(translate_units)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = WCS Keywords

Number of WCS axes: 2
CTYPE : 'WAVE'  'RA'  
CRVAL : 0.002919277457  83.81042  
CRPIX : 32768.0  1.0  
PC1_1 PC1_2  : 1.0  0.0  
PC2_1 PC2_2  : 0.0  1.0  
CDELT : 6.630101933e-09  1.0  
NAXIS : 0  0
translate_units = '', naxis = None

    def fix(self, translate_units='', naxis=None):
        """…"""
        if self.wcs is not None:
            self._fix_scamp()
            fixes = self.wcs.fix(translate_units, naxis)
            for key, val in fixes.items():
                if val not in ("No change", "Success"):
                    if (key == 'datfix' and '1858-11-17' in val and
                            not np.count_nonzero(self.wcs.mjdref)):
                        continue
>                   warnings.warn(
                        ("'{0}' made the change '{1}'.").
                        format(key, val),
                        FITSFixedWarning)
E                   astropy.wcs.wcs.FITSFixedWarning: 'datfix' made the change 'Set MJD-OBS to 53925.853472 from DATE-OBS'.

astropy/wcs/wcs.py:706: FITSFixedWarning
```
and
```

    def test_dict_init():
        """
        Test that WCS can be initialized with a dict-like object
        """
    
        # Dictionary with no actual WCS, returns identity transform
        with pytest.warns(None) as wrng:
            w = wcs.WCS({})
        _check_v71_dateref_warnings(wrng)
    
        xp, yp = w.wcs_world2pix(41., 2., 1)
    
        assert_array_almost_equal_nulp(xp, 41., 10)
        assert_array_almost_equal_nulp(yp, 2., 10)
    
        # Valid WCS
        hdr = {
            'CTYPE1': 'GLON-CAR',
            'CTYPE2': 'GLAT-CAR',
            'CUNIT1': 'deg',
            'CUNIT2': 'deg',
            'CRPIX1': 1,
            'CRPIX2': 1,
            'CRVAL1': 40.,
            'CRVAL2': 0.,
            'CDELT1': -0.1,
            'CDELT2': 0.1
        }
        if _WCSLIB_VER >= Version('7.1'):
            hdr['DATEREF'] = '1858-11-17'
    
>       w = wcs.WCS(hdr)

astropy/wcs/tests/test_wcs.py:216: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/wcs/wcs.py:527: in __init__
    self.fix(translate_units=translate_units)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = WCS Keywords

Number of WCS axes: 2
CTYPE : 'GLON-CAR'  'GLAT-CAR'  
CRVAL : 40.0  0.0  
CRPIX : 1.0  1.0  
PC1_1 PC1_2  : 1.0  0.0  
PC2_1 PC2_2  : 0.0  1.0  
CDELT : -0.1  0.1  
NAXIS : 0  0
translate_units = '', naxis = None

    def fix(self, translate_units='', naxis=None):
        """…"""
        if self.wcs is not None:
            self._fix_scamp()
            fixes = self.wcs.fix(translate_units, naxis)
            for key, val in fixes.items():
                if val not in ("No change", "Success"):
                    if (key == 'datfix' and '1858-11-17' in val and
                            not np.count_nonzero(self.wcs.mjdref)):
                        continue
>                   warnings.warn(
                        ("'{0}' made the change '{1}'.").
                        format(key, val),
                        FITSFixedWarning)
E                   astropy.wcs.wcs.FITSFixedWarning: 'datfix' made the change 'Set MJDREF to 0.000000 from DATEREF'.

astropy/wcs/wcs.py:706: FITSFixedWarning
```
and
```
________________________________ test_validate _________________________________

    @pytest.mark.filterwarnings("ignore")
    def test_validate():
        results = wcs.validate(get_pkg_data_filename("data/validate.fits"))
        results_txt = sorted(set([x.strip() for x in repr(results).splitlines()]))
        version = wcs._wcs.__version__
        if version[0] in ['6', '7']:
            filename = 'data/validate.6.txt'
        elif version[0] == '5':
            if version >= '5.13':
                filename = 'data/validate.5.13.txt'
            else:
                filename = 'data/validate.5.0.txt'
        else:
            filename = 'data/validate.txt'
        with open(get_pkg_data_filename(filename), "r") as fd:
            lines = fd.readlines()
>       assert sorted(set([x.strip() for x in lines])) == results_txt
E       assert ['', "'HZ' ->... system", ...] == ['', "'HZ' ->... units:", ...]
E         At index 4 diff: "- 'unitfix' made the change 'Changed units:" != "- 'datfix' made the change 'Set MJD-OBS to 55007.362083 from DATE-"
E         Right contains 2 more items, first extra item: 'parameters are also present'
E         Use -v to get the full diff

astropy/wcs/tests/test_wcs.py:446: AssertionError
```

**Comment by olebole** [comment]
Ooops, I am sorry. I just got confused myself by not using the current version of the Astropy package (that happens when one works on several machines; just forgot the `git pull` for the package).
These problems were already solved by #11260 resp. in the Debian package astropy_4.2-4.
Closing this...
